### PR TITLE
Create objectdatasource-assembly-not-recognized-by-standalone-designe…

### PR DESCRIPTION
**Pull request by todorarabadzhiev**
…r.md

### CODE REVIEW

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: ObjectDataSource Assembly Not Recognized by the Standalone Designer
```
Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: Loaded

Comment (praykov) [reply knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: ObjectDataSource Assembly Not Recognized by the Standalone Designer

Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: Assembly Used in ObjectDataSource is not Loaded by the Standalone Designer

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+description: How to troubleshoot which assemblies to add to the Standalone designer folder when an ObjectDataSource is not property recognized
```
Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: How to troubleshoot which assemblies to add to the Standalone Designer folder when an ObjectDataSource assembly types are missing in the wizard.

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+type: troubleshooting
+page_title: 
+slug: objectdatasource-assembly-not-recognized-by-standalone-designer
```
Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: loaded

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+When you use an ObjectDataSourcem, which DataSource has external dependencies, the Standalone designer may not recognize the main ObjectDataSource assembly even when it is properly added to the designer folder and registered in its configuration file.
```
Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: The Standalone Designer may not recognize the main ObjectDataSource assembly (even though the assembly is properly added to the designer folder and registered in its configuration file) when this assembly depends on other assemblies.

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Cause\Possible Cause(s)
+The reason is that the Standalone designer cannot find the external assemblies when trying to get the data source schema.
+
+## Solution
+You may use [Fuslogvw](https://docs.microsoft.com/en-us/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer) to investigate the issue. The idea is to see what dependencies the designer tries to load and eventually which ones cannot be found.
```
Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: which dependencies?

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+Here we demonstrate the approach step-by-step with an ObjectDataSource that utilizes EntityFrameworkCore version 3.1.1.
+
+1. Start [Fuslogvw](https://techcommunity.microsoft.com/t5/iis-support-blog/fusion-log-viewer-fuslogvw-exe/ba-p/784396) in [Developer Command Prompt for Visual Studio](https://docs.microsoft.com/en-us/dotnet/framework/tools/developer-command-prompt-for-vs) - screenshot [StartFuslogvw.png](https://www.telerik.com/docs/default-source/knowledgebasearticleattachments/reporting/startfuslogvw.png?sfvrsn=e57b63f_2)
+2. Setup Fuslogvw to record the failures in a custom folder, for example, 'c:\temp' - screenshot [SetupFuslogvw.png](https://www.telerik.com/docs/default-source/knowledgebasearticleattachments/reporting/setupfuslogvw.png?sfvrsn=d915da24_2)
+3. Add the assembly (e.g. _ClassLibrary2_) with the DataSource/DataMember of the ObjectDataSource in the folder of the Standalone designer and register it in the designer configuration file - [Extending Report Designer](../standalone-report-designer-extending-configuration)
```
Comment (praykov) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: in the Standalone Report Designer folder?

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+4. Open the Standalone designer and start the ObjectDataSource wizard to configure it. If _ClassLibrary2_ is not discovered by the wizard, as shown in screenshot [DataSourcesNotFound.png](https://www.telerik.com/docs/default-source/knowledgebasearticleattachments/reporting/datasourcesnotfound.png?sfvrsn=9efca29c_2), the Fuslogvw should have logged a failure in the '(_specified folder_)\Default\Telerik.ReportDesigner.exe', in our case 'C:\temp\Default\Telerik.ReportDesigner.exe'. The particular log file is _Microsoft.EntityFrameworkCore, Version=3.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60.HTM_ and indicates that this assembly was not found
```
Comment (nelci592) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: here you may include a link for the ObjectDataSource wizard article: https://docs.telerik.com/reporting/objectdatasource-wizard

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+## Description
+When you use an ObjectDataSourcem, which DataSource has external dependencies, the Standalone designer may not recognize the main ObjectDataSource assembly even when it is properly added to the designer folder and registered in its configuration file.
+
+## Cause\Possible Cause(s)
+The reason is that the Standalone designer cannot find the external assemblies when trying to get the data source schema.
+
+## Solution
+You may use [Fuslogvw](https://docs.microsoft.com/en-us/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer) to investigate the issue. The idea is to see what dependencies the designer tries to load and eventually which ones cannot be found. 
+Here we demonstrate the approach step-by-step with an ObjectDataSource that utilizes EntityFrameworkCore version 3.1.1.
```
Comment (nelci592) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: you may include a link for the object data source https://docs.telerik.com/reporting/objectdatasource

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+6. Close the designer and repeat steps 4-6 until the assembly is discovered by the designer.
+
+Note that there may be other failures logged by Fuslogvw. They are related to internal designer usage - you don't need to add them. The [FuslogvwLogs.zip](https://www.telerik.com/docs/default-source/knowledgebasearticleattachments/reporting/fuslogvwlogs.zip?sfvrsn=cfa37cc5_2) file contains the ObjectDataSource-relevant logs for the example. A screenshot of the entire log folder can be seen on [FuslogvwForReportDesigner.png](https://www.telerik.com/docs/default-source/knowledgebasearticleattachments/reporting/fuslogvwforreportdesigner.png?sfvrsn=43802be5_2).
```
Comment (KatGitHub) [inline knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: May be to rephrase like this: Note that there may be other failures to discover assemblies logged by Fuslogvw. Those assemblies are used internally by the designer - you don't need to add them.

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md (hunk 8 above)
Comment (todorarabadzhiev) [reply knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: Included in the start

File: knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md (hunk 7 above)
Comment (todorarabadzhiev) [reply knowledge-base/objectdatasource-assembly-not-recognized-by-standalone-designer.md]: included

**Pull request merged**